packages/inventory/src/constants.js

```diff
@@ -0,0 +1,68 @@
+export const TEXT_FILTER = 'hostname_or_id';
+
+export function constructValues(groupValue, tags) {
+    return Object.entries(groupValue).map(([ key, value ]) => {
+        if (value) {
+            const { tagName, tagValue } = tags[key];
+            return {
+                key,
+                name: `${tagName} - ${tagValue}`
+            };
+        }
+    }).filter(Boolean);
+}
+
+export function mapGroups(currSelection, allTags, valuesKey = 'values') {
+    return Object.entries(currSelection).map(([ groupKey, groupValue ]) => {
+        const { tags, name } = allTags[groupKey];
+        const values = constructValues(groupValue, tags);
+        if (values && values.length > 0) {
+            return {
+                type: 'tags',
+                key: groupKey,
+                category: name,
+                [valuesKey]: values
+            };
+        }
+    }).filter(Boolean);
+}
+
+export function filterToGroup(filter = [], valuesKey = 'values') {
+    return filter.reduce((accGroup, group) => ({
+        ...accGroup,
+        [group.key]: group[valuesKey].reduce((acc, curr) => ({ ...acc, [curr.key]: true }), {})
+    }), {});
+}
+
+export function constructGroups(allTags) {
+    return allTags.map(({ name, tags }, key) => ({
+        label: name,
+        value: key,
+        type: 'checkbox',
+        items: tags.map(({ tagName, tagValue }) => ({
+            label: `${tagName} - ${tagValue}`
+        }))
+    }));
+}
+
+export function reduceFilters(filters) {
+    return filters.reduce((acc, oneFilter) => {
+        if (oneFilter.value === TEXT_FILTER) {
+            return { ...acc, textFilter: oneFilter.filter };
+        } else if (oneFilter.hasOwnProperty('tagFilters')) {
```

Comment:
Since user can have any keys as the object attribute, maybe it would be better to use `Object.hasOwnProperty` to prevent possible runtime crash.

Reply:
I don't really like `Object.hasOwnProperty` because it's so long, however [JS#Operators/in](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in) could be better here.

Reply:
Works to